src/test/java/uk/gov/pay/api/it/validation/telephone/CreateTelepehonePaymentLastFourCardDigitsValidation.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.pay.api.it.validation.telephone;
+
+import org.junit.Before;
+import org.junit.Test;
+import uk.gov.pay.api.resources.telephone.TelephonePaymentResourceITBase;
+import uk.gov.pay.api.utils.PublicAuthMockClient;
+
+import java.util.Map;
+
+
+public class CreateTelepehonePaymentLastFourCardDigitsValidation extends TelephonePaymentResourceITBase {
+
+    private PublicAuthMockClient publicAuthMockClient = new PublicAuthMockClient(publicAuthMock);
+
+    @Before
+    public void setUpBearerToken() {
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID);
+    }
+
+    @Test
+    public void respondWith422_whenThreeDigitsProvidedOnly() {
+        String payload = toJson(Map.of("amount", 100,
+                "reference", "Some reference",
+                "description","hi",
+                "processor_id", "1PROC",
+                "provider_id", "1PROV",
+                "card_type", "visa",
+                "card_expiry", "01/99",
+                "last_four_digits", "123",
+                "first_six_digits", "123456"));
+
+        postPaymentResponse(payload).statusCode(422);
+    }
+
+    @Test
+    public void respondWith422_whenFiveDigitsProvidedOnly() {
+        String payload = toJson(Map.of("amount", 100,
+                "reference", "Some reference",
+                "description","hi",
+                "processor_id", "1PROC",
+                "provider_id", "1PROV",
+                "card_type", "visa",
+                "card_expiry", "01/99",
+                "last_four_digits", "12345",
+                "first_six_digits", "123456"));
+
+        postPaymentResponse(payload).statusCode(422);
+    }
+    
+    @Test
+    public void respondWith422_whenNullProvided() {
+        String payload = "{" +
+                "  \"amount\" : 100," +
+                "  \"reference\" : \"Some reference\"," +
+                "  \"description\" : \"hi\"," +
+                "  \"processor_id\" : \"1PROC\"," +
+                "  \"provider_id\" : \"1PROV\"," +
+                "  \"card_type\" : \"visa\"," +
+                "  \"card_expiry\" : \"01/99\"," +
+                "  \"last_four_digits\" : null," +
+                "  \"first_six_digits\" : \"123456\"" +
+                "}";
+
+        postPaymentResponse(payload).statusCode(422);
+    }
```

Comment:
I think all these tests can ResourceTests
https://www.dropwizard.io/1.3.11/docs/manual/testing.html
This would mean you wouldn't actually have to so spin up the app, so the tests would be much faster.

Reply:
These have been added